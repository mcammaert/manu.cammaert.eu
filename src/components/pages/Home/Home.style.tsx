import styled from 'styled-components';

export const Home = styled.div`
  position: absolute;
`;

export default { Home };
