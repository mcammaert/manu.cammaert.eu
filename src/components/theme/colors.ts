export const colors = {
  primary90: '#0c2d48',
  primary80: '#0b395d',
  primary70: '#0b4573',
  primary60: '#0d5189',
  primary50: '#145da0',
  primary40: '#367bb1',
  primary30: '#5c99c1',
  primary20: '#85b7d0',
  primary10: '#b1d4e0',

  secondary90: '#ca3f36',
  secondary80: '#d35239',
  secondary70: '#db633e',
  secondary60: '#e37443',
  secondary50: '#eb844a',
  secondary40: '#f79b77',
  secondary30: '#fdb5a1',
  secondary20: '#fdd0c8',
  secondary10: '#f7eceb',

  accent1: '#d45e95',
  accent2: '#b290d4',
  accent3: '#97b8e6',

  // 10 shades of grey
  black: '#1E2025',
  charcoal: '#262830',
  shadow: '#2E313A',
  lead: '#585965',
  steel: '#787C8A',
  gray: '#989DAC',
  smoke: '#BEC2C9',
  fog: '#EDEEF0',
  white: '#F6F7F9',
  snow: '#FCFCFC',

  // slate
  slateLight: '#3279AD',
  slate: '#4C6E89',
  slateDark: '#455C77',

  // teal
  teal: '#3D829C',
  tealDark: '#36768B',

  // laguna
  lagunaLight: '#4FB8F3',
  laguna: '#57A7E3',
  lagunaDark: '#3680DC',

  // warm colors
  red: '#F60C5B',
  orange: '#F86C5C',
  sunflower: '#EDC172',
  lemon: '#FFE362',

  // purples
  grape: '#374278',
  plum: '#725FA4',
  lavender: '#936FBA',
  purple: '#C163C6',

  // greens
  forest: '#50C04E',
  lawn: '#86DE6A',
  aquamarine: '#14DAB5',
  emerald: '#0DCACE',
  turquoise: '#57D5D6',

  // browns
  chocolate: '#3C322D',
  barnwood: '#5E5E5F',
};

export default colors;
