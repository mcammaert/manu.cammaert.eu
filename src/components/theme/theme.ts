import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body {
    background-color: #ccc;
  }
`;

const theme = {
  main: 'mediumseagreen',
};

export default theme;
