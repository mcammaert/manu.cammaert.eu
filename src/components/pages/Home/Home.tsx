import React, { useEffect, useState } from 'react';

import { useInterval } from 'react-use';
import { Link } from 'react-router-dom';

import { Container } from 'components/atoms/Container';
import { Copy } from 'components/atoms/Copy';
import { Default } from 'components/templates/Default';

import { HomeProps } from './Home.types';

import S from './Home.style';

const getSalutation = (hour: number): string => {
  if (hour <= 6) return 'Goede nacht';
  if (hour <= 10) return 'Goede morgen';
  if (hour <= 14) return 'Goede middag';
  if (hour <= 17) return 'Goede namiddag';
  if (hour <= 21) return 'Goede avond';
  return 'Goede nacht';
};

const Home: React.FC<HomeProps> = () => {
  const [hour, setHour] = useState(0);

  useInterval(() => {
    setHour(new Date(Date.now()).getHours());
  }, 10000);

  useEffect(() => {
    setHour(new Date(Date.now()).getHours());
  }, []);

  return (
    <S.Home>
      <Default>
        <Container>
          <Copy margin size={1}>
            {getSalutation(hour)}, ik ben Manu Cammaert, front-end developer / designer met een passie voor
            <span> </span>
            <abbr title="User Interface">UI</abbr>
            <span> </span>
            design en fotografie.
          </Copy>
          <Copy margin>
            Mijn specialisatie ligt momenteel bij React, waarin ook deze website is gebouwd. Je kan de
            <span> </span>
            <a href="https://github.com/mcammaert/manu.cammaert.eu" aria-label="broncode van deze website">
              broncode ervan
            </a>
            <span> </span>
            hier bekijken.
          </Copy>
          <Copy>
            Bekijk hier
            <span> </span>
            <Link to="/cv">
              mijn <abbr title="curriculum vitae">cv</abbr>
            </Link>
            <span> </span>
            of <Link to="/contact">contacteer me</Link> voor vragen.
          </Copy>
        </Container>
      </Default>
    </S.Home>
  );
};

export default Home;
