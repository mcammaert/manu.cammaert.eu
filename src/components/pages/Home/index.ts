import * as S from './Home.style';

export { default as Home } from './Home';
export * from './Home.types';
export { S };
