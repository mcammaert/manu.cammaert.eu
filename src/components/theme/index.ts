import { createGlobalStyle } from 'styled-components';
import 'normalize.css';
import { colors } from './colors';

export interface GlobalStyleProps {
  readonly light: boolean;
}

// const backgroundColor = (location: string) => (location === '/' ? colors.oxfordBlue : colors.fuzzyWuzzy);

export const GlobalStyle = createGlobalStyle<GlobalStyleProps>`
  body {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    font-family: 'Nunito Sans', sans-serif;
    width: 100vw;
    background: ${({ light }) => (light ? colors.white : colors.black)};
    background-attachment: fixed;    
    color: ${({ light }) => (light ? colors.black : colors.snow)};
    transition: all 800ms;
    a {
      color: ${({ theme }) => theme.colors.secondary};
    }
  }
`;

export const lightTheme = {
  margin: 15,
  colors: {
    primary: colors.charcoal,
    primaryInvert: colors.snow,
    secondary: colors.primary60,
    secondaryInvert: colors.white,
    tertiary: colors.secondary80,
    tertiaryInvert: colors.white,
    copy: colors.charcoal,
    copyInvert: colors.white,
  },
};

export const darkTheme = {
  margin: 15,
  colors: {
    primary: colors.snow,
    primaryInvert: colors.black,
    secondary: colors.primary30,
    secondaryInvert: colors.primary90,
    tertiary: colors.secondary30,
    tertiaryInvert: colors.secondary80,
    copy: colors.white,
    copyInvert: colors.charcoal,
  },
};
