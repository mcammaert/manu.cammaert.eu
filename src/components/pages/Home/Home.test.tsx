import React from 'react';

import { cleanup } from '@testing-library/react';
import 'jest-styled-components';

import { renderComponent } from 'testing/react-testing-library';
import { lightTheme } from 'components/theme';

import Home from './Home';

afterEach(cleanup);

describe('Home', () => {
  describe('default props', () => {
    it('should match the snapshot', () => {
      const component = <Home />;
      const { container } = renderComponent({ theme: lightTheme, component });
      expect(container).toMatchSnapshot();
    });
  });
});
