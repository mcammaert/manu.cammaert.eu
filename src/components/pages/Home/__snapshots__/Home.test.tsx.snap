// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home default props should match the snapshot 1`] = `
.c1 {
  width: 100vw;
}

.c2 {
  position: absolute;
  font-size: 0.7501875468867217em;
  padding-top: 5.610232974924883em;
  padding-bottom: 5.610232974924883em;
  width: 100vw;
}

.c5 {
  color: rgba(38,40,48,0.45);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 0;
}

.c4 {
  width: 100vw;
  max-width: 800px;
  padding-left: 0.9375rem;
  padding-right: 0.9375rem;
  margin-top: 0;
  margin-bottom: 0;
}

.c0 {
  position: absolute;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <p>
        Home
      </p>
      <footer
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              Copyright Â© 
              2019
               Manu Cammaert, alle rechten voorbehouden.
            </div>
          </div>
        </div>
      </footer>
    </div>
  </div>
</div>
`;
